### PR TITLE
[app] Add icon property example

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[app] Add icon property example</changelog>`.

### Summary of changes
This pr adds an example displaying multiple images with a SymbolLayer, fellows the example in [iOS](https://docs.mapbox.com/ios/maps/examples/multiple-images/)
image

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/IconPropertyActivity.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        // This tileset was created by uploading NPS shapefiles to Mapbox Studio.
+        +vectorSource(SOURCE_ID) {
+          url(SOURCE_URI)
+        }
+        // Create a symbol layer and access the layer contained.
+        +symbolLayer(LAYER_ID, SOURCE_ID) {
+          // Expression that adds conditions to the source to determine styling.
+          // `POITYPE` refers to a key in the data source. The values tell us which icon to use from the sprite sheet
+          iconImage(
+            Expression.switchCase {
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/IconPropertyActivity.kt:47]:
```suggestion
           switchCase {
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/IconPropertyActivity.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        +image(PICNIC_AREA) {
+          bitmap(BitmapFactory.decodeResource(resources, R.drawable.nps_picnic_area))
+        }
+        // Access a vector tileset that contains places of interest at Yosemite National Park.
+        // This tileset was created by uploading NPS shapefiles to Mapbox Studio.
+        +vectorSource(SOURCE_ID) {
+          url(SOURCE_URI)
+        }
+        // Create a symbol layer and access the layer contained.
+        +symbolLayer(LAYER_ID, SOURCE_ID) {
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/IconPropertyActivity.kt:43]: missing SourceLayer property here:
```suggestion
        +symbolLayer(LAYER_ID, SOURCE_ID) {
          // Access the layer that contains the Point of Interest (POI) data.
          // The source layer property is a unique identifier for a layer within a vector tile source.
          sourceLayer("Yosemite_POI-38jhes")
```

**Review by pengdev — APPROVED** [review]
LGTM

**Review by kiryldz — COMMENTED** [review]
@Chaoba I think this example is a better example of `AddMarkersSymbolActivity` - guess we can delete it then, WDYT?

**Comment by Chaoba** [comment]
> @Chaoba I think this example is a better example of `AddMarkersSymbolActivity` - guess we can delete it then, WDYT?

`AddMarkersSymbolActivity` demonstrate how to `addStringProperty` programmed, which is included in this demo. So IMO it's worth keeping them both.

**Review by kiryldz — APPROVED** [review]

**Pull request merged**